data/scripting/infrastructure.lua

```diff
@@ -259,19 +259,24 @@ end
 --    :type waretable_rocks: :const:`table`
 --    :arg waretable_trees: a table of pairs {:const:`"name"` = value} to add if no trees were found nearby
 --    :type waretable_trees: :const:`table`
+--    :arg min_trees: minimal number of trees (optional)
+--    :type min_trees: :class:`integer`
 
-function check_trees_rocks_poor_hamlet(player, sf, warehouse, waretable_rocks, waretable_trees)
-   -- NOTE: pessimistically, this could be a single rock and a single tree
+function check_trees_rocks_poor_hamlet(player, sf, warehouse, waretable_rocks, waretable_trees, min_trees)
+   if not min_trees then -- pass argument if this default does not match
+      min_trees = waretable_trees["log"] + (waretable_trees["planks"] or 0) * 2
+   end
+   -- NOTE: pessimistically, this could be a single rock
    local has_rocks = false
-   local has_trees = false
+   local n_trees = 0
```

Comment:
as the bool variable here is now an integer with a changed name please adapt the condition in line 276 to reflect this

Reply:
Oops, not using an undefined variable makes sense of course.
It worked anyway. My code was counting on the trees until enough rocks were found. The final conditions were correct, so the result was as expected.